internal/provider/providers/scaleway/provider.go

```diff
@@ -0,0 +1,188 @@
+	"github.com/qdm12/ddns-updater/internal/provider/utils"
+	"github.com/qdm12/ddns-updater/pkg/publicip/ipversion"
+)
+
+type Provider struct {
+	domain 		string
+	owner  		string
+	ipVersion  	ipversion.IPVersion
+	ipv6Suffix 	netip.Prefix
+	secretKey   string
+	field_name	string
+	ttl		 	uint16
+}
+
+func New(data json.RawMessage, domain, owner string,
+	ipVersion ipversion.IPVersion, ipv6Suffix netip.Prefix) (
+	provider *Provider, err error,
+) {
+	var providerSpecificSettings struct {
+        SecretKey	string	`json:"secret_key"`
+		FieldName	string	`json:"field_name"`
+		TTL			uint16	`json:"ttl"`
+    }
+	err = json.Unmarshal(data, &providerSpecificSettings)
+	if err != nil {
+		return nil, fmt.Errorf("json decoding provider specific settings: %w", err)
+	}
+
+	if providerSpecificSettings.TTL == 0 {
+        providerSpecificSettings.TTL = 3600
+    }
+
+	err = validateSettings(domain,
+		providerSpecificSettings.SecretKey)
+	if err != nil {
+		return nil, fmt.Errorf("validating provider specific settings: %w", err)
+	}
+
+	return &Provider{
+		domain:     domain,
+		owner:      owner,
+		ipVersion:  ipVersion,
+		ipv6Suffix: ipv6Suffix,
+        secretKey:  providerSpecificSettings.SecretKey,
+		field_name: providerSpecificSettings.FieldName,
+		ttl:		providerSpecificSettings.TTL,
+	}, nil
+}
+
+func validateSettings(domain, secretKey string) (err error) {
+	err = utils.CheckDomain(domain)
+	if err != nil {
+		return fmt.Errorf("%w: %w", errors.ErrDomainNotValid, err)
+	}
+
+	switch {
+    case secretKey == "":
+        return fmt.Errorf("%w", errors.ErrSecretKeyNotSet)
+    }
+
+	return nil
+}
+
+func (p *Provider) String() string {
+	return utils.ToString(p.domain, p.owner, constants.Dyn, p.ipVersion)
+}
+
+func (p *Provider) Domain() string {
+	return p.domain
+}
+
+func (p *Provider) Owner() string {
+	return p.owner
+}
+
+func (p *Provider) IPVersion() ipversion.IPVersion {
+	return p.ipVersion
+}
+
+func (p *Provider) IPv6Suffix() netip.Prefix {
+	return p.ipv6Suffix
+}
+
+func (p *Provider) Proxied() bool {
+	return false
+}
+
+func (p *Provider) BuildDomainName() string {
+	return utils.BuildDomainName(p.owner, p.domain)
+}
+
+func (p *Provider) HTML() models.HTMLRow {
+	return models.HTMLRow{
+		Domain: fmt.Sprintf("<a href=\"http://%s\">%s</a>", p.BuildDomainName(), p.BuildDomainName()),
+		Owner:  p.Owner(),
+		Provider:  "<a href=\"https://www.scaleway.com/\">Scaleway</a>",
+		IPVersion: p.ipVersion.String(),
+	}
+}
+
+// Update updates the DNS record for the domain using Scaleway's API.
+// See: https://www.scaleway.com/en/developers/api/domains-and-dns/#path-records-update-records-within-a-dns-zone
+func (p *Provider) Update(ctx context.Context, client *http.Client, ip netip.Addr) (newIP netip.Addr, err error) {
+    u := url.URL{
+        Scheme: "https",
+        Host:   "api.scaleway.com",
+        Path:   fmt.Sprintf("/domain/v2beta1/dns-zones/%s/records", p.domain),
+    }
+
+	field_type := "A"
```

Comment:
nit rename to `fieldType` (Go convention is no snake_case 🤷  Only camelCase)

Reply:
true! Should have noticed... changed accordingly!